### PR TITLE
Added token revocation for REST API

### LINKED ISSUE #30820: Token Revocation REST API
### Description (*)
As a user, when I log out, I want to guarantee that my token is revoked. Currently, with the REST API, there's no known way to do this.

The GraphQL api supports this:
https://devdocs.magento.com/guides/v2.3/graphql/mutations/revoke-customer-token.html

### Expected behavior (*)
I can securely revoke my authentication token.

### Benefits
Basic security improvement for customers.

### Additional information
cc: @zetlen

### PR DESCRIPTION
**Pull request by ProkopovVitaliy**
### Description (*)

Created new  token revocation endpoint for REST API: ` /V1/integration/customer/revoke-customer-token`. 
Feature have similar functionality like here https://devdocs.magento.com/guides/v2.3/graphql/mutations/revoke-customer-token.html. 

### Fixed Issues (if relevant)

1. Fixes magento/magento2#30820

### Manual testing scenarios (*)

1. Generate customer authorization token.
2. Send request to new created `/V1/integration/customer/revoke-customer-token` endpoint
3. Check if token was removed

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by ProkopovVitaliy** [comment]
@magneto run all tests

**Comment by ProkopovVitaliy** [comment]
@magento run all test

**Comment by ProkopovVitaliy** [comment]
@magento run all tests

**Comment by ProkopovVitaliy** [comment]
@magento run Functional Tests B2B

**Comment by ProkopovVitaliy** [comment]
@magento run Functional Tests B2B

**Comment by ProkopovVitaliy** [comment]
@magento run all tests

**Comment by ProkopovVitaliy** [comment]
@magento  run Integration Tests

**Comment by ProkopovVitaliy** [comment]
@magento run Functional Tests B2B

File: dev/tests/api-functional/testsuite/Magento/Customer/Api/AccountManagementRevokeCustomerTokenTest.php [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+            'soap' => [
+                'service' => self::INTEGRATION_SERVICE,
+                'serviceVersion' => self::SERVICE_VERSION,
+                'operation' => self::INTEGRATION_SERVICE . 'RevokeCustomerAccessToken',
+                'token' => $token,
+            ]
+        ];
+
+        $requestData = [];
+        if (TESTS_WEB_API_ADAPTER === 'soap') {
```
Comment (eduard13) [inline dev/tests/api-functional/testsuite/Magento/Customer/Api/AccountManagementRevokeCustomerTokenTest.php]:
```suggestion
        if (TESTS_WEB_API_ADAPTER === self::ADAPTER_SOAP) {
```

File: dev/tests/api-functional/testsuite/Magento/Customer/Api/AccountManagementRevokeCustomerTokenTest.php [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+    /**
+     * Test token revoking for guest customer
+     */
+   public function testRevokeCustomerTokenForGuestCustomer()
+    {
+        $this->expectException(Exception::class);
+        $requestData = [];
+
+        if (TESTS_WEB_API_ADAPTER === 'soap') {
```
Comment (eduard13) [inline dev/tests/api-functional/testsuite/Magento/Customer/Api/AccountManagementRevokeCustomerTokenTest.php]:
```suggestion
        if (TESTS_WEB_API_ADAPTER === self::ADAPTER_SOAP) {
```

File: dev/tests/api-functional/testsuite/Magento/Customer/Api/AccountManagementRevokeCustomerTokenTest.php [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+    const RESOURCE_PATH = '/V1/integration/customer/revoke-customer-token';
+    const INTEGRATION_SERVICE = 'integrationCustomerTokenServiceV1';
+    const SERVICE_VERSION = 'V1';
+
+    /**
+     * Test token revoking for authenticated customer
+     *
+     * @magentoApiDataFixture Magento/Customer/_files/customer.php
+     */
+    public function testRevokeCustomerToken()
```
Comment (eduard13) [inline dev/tests/api-functional/testsuite/Magento/Customer/Api/AccountManagementRevokeCustomerTokenTest.php]:
```suggestion
    public function testRevokeCustomerToken(): void
```

File: dev/tests/api-functional/testsuite/Magento/Customer/Api/AccountManagementRevokeCustomerTokenTest.php [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+        /** @var CustomerTokenServiceInterface $customerTokenService */
+        $customerTokenService = ObjectManager::getInstance()->get(CustomerTokenServiceInterface::class);
+
+        return $customerTokenService->createCustomerAccessToken($userName, $password);
+    }
+
+    /**
+     * Test token revoking for guest customer
+     */
+   public function testRevokeCustomerTokenForGuestCustomer()
```
Comment (eduard13) [inline dev/tests/api-functional/testsuite/Magento/Customer/Api/AccountManagementRevokeCustomerTokenTest.php]:
```suggestion
   public function testRevokeCustomerTokenForGuestCustomer(): void
```

**Review by eduard13 — CHANGES_REQUESTED** [review]
Hi @ProkopovVitaliy please find below couple minor suggestions.

Thank you.

**Comment by ProkopovVitaliy** [comment]
@magento run all tests

**Comment by ProkopovVitaliy** [comment]
@magento run Functional Tests CE

**Comment by ProkopovVitaliy** [comment]
@magento run WebAPI Tests

**Comment by ProkopovVitaliy** [comment]
@magento run Functional Tests B2B

**Comment by ProkopovVitaliy** [comment]
@magento run WebAPI Tests, Functional Tests B2B

**Comment by eduard13** [comment]
@magento run WebAPI Tests, Functional Tests B2B

**Comment by ProkopovVitaliy** [comment]
@eduard13 Code review recommendations implemented, tests passed.

**Review by eduard13 — APPROVED** [review]
Thank you for this cool feature 👍.

**Comment by magento-engcom-team** [comment]
Hi @eduard13, thank you for the review.
[ENGCOM-8937](https://jira.corp.magento.com/browse/ENGCOM-8937) has been created to process this Pull Request

**Comment by engcom-Oscar** [comment]
:heavy_check_mark: QA passed

Was checked case from Manual testing scenarios 
1. Generate customer authorization token.
2. Send request to new created `/V1/integration/customer/revoke-customer-token` endpoint
3. Check if token was removed

:heavy_check_mark: **After:**
Customer token was successfully revoked
Screenshot from 2021-03-24 17-30-44

**Comment by ihor-sviziev** [comment]
Hi @sidolov,
Adding a new endpoint shouldn't be treated as a backward-incompatible change, and we can accept it. Right?

**Comment by sidolov** [comment]
@ihor-sviziev as far as I remember - yes, let's have the same solution for 2.4-develop and check the SVC build

**Review by ihor-sviziev — CHANGES_REQUESTED** [review]
Hi @ProkopovVitaliy,
We want to focus more on 2.4 release line.
Could you rebase your changes on top of 2.4-develop branch?

**Comment by eduard13** [comment]
@magento run all tests

**Comment by ihor-sviziev** [comment]
@magento run Database Compare, Functional Tests CE, Functional Tests EE, Sample Data Tests B2B, Static Tests, Unit Tests

**Review by eduard13 — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @eduard13, thank you for the review.
[ENGCOM-8937](https://jira.corp.magento.com/browse/ENGCOM-8937) has been created to process this Pull Request

**Comment by eduard13** [comment]
Hi @sidolov, it seems that we are fine here, all the builds are green as well for 2.4-develop.

**Review by ihor-sviziev — APPROVED** [review]
Approved ✔

Note: this PR was already tested in https://github.com/magento/magento2/pull/32481#issuecomment-805925546, but should be re-tested again

**Comment by magento-engcom-team** [comment]
Hi @ihor-sviziev, thank you for the review.
[ENGCOM-8937](https://jira.corp.magento.com/browse/ENGCOM-8937) has been created to process this Pull Request

**Comment by engcom-Alfa** [comment]
:heavy_check_mark: QA Passed

**Preconditions:**

1. Have Magento latest installed


**Manual testing scenario:**

1. Got the customer token using request /V1/integration/customer/token

2. Cross checked using received token with another request /V1/carts/mine; Just to make sure the token is working!
 
3. post requested - V1/integration/customer/revoke-customer-token; Validated the response


**Before:** :heavy_multiplication_x: There was no revoke-customer-token API request at all

before


**After:** :heavy_check_mark: revoke-customer-token introduced working absolutely fine

after


After the revoke-customer-token re-tried to access the /V1/carts/mine request and identified that the revoked token is no more valid!

This PR is relevant to API request and has no impact functionally. There is an auto-test covered in the PR and no hence additional regression is required

**Comment by ishakhsuvarov** [comment]
@magento run all tests

**Comment by ishakhsuvarov** [comment]
@magento run Integration Tests

**Pull request merged**